Python27 compatibility (mostly) and PEP8 compliance (mostly)

kili-ilo opened the pull request:
Lots of re-organizing imports (not in all files... yet) and wrapping lines at 80 characters

kili-ilo committed:
Fix data_stats mocking, reformat comments and math

kili-ilo committed:
Refactor all test imports to top of file

kili-ilo committed:
Make a few PEP8 fixes

kili-ilo committed:
Shorten some long lines

kili-ilo committed:
Shorten some long lines in scripts

kili-ilo committed:
Revert data_stats comparisons because tests

I stupidly forgot to run tests after changing
a few "if x == []"s to "if not x"s

kili-ilo committed:
Add requirements.txt

kili-ilo committed:
Make repo Python27-compatible

Just pass tests and include
"from __future__ import print_function",
but not all code has been run with Python 2.7

jvanbrug commented:
Make sure to resolve the handful of TODOs I added

jvanbrug commented:
Most importantly, @fictivelogic what is the `git` library that we import? Is it on the Python Packaging Index? I'm thinking we should remove it to simplify our requirements, and it's not yet in the requirements text files.

fictivelogic committed:
Fix bug in Python2.7 requirements file

fictivelogic committed:
Update python34.txt requirements

Add gitpython package and its dependencies. Tidy up a little.

fictivelogic committed:
Change empty lists to empty numpy array

For type consistency. Note that the dtypes of the arrays
are NOT specified in the __init__(self) but rather are
specified in the init_movie_and_user_arrays(self) function.

fictivelogic committed:
Update data stats empty list checks

fictivelogic committed:
Fix divide by zero runtime error
